data_model/src/expression.rs

```diff
@@ -49,12 +49,9 @@ pub struct EvaluatesTo<V: TryFrom<Value>> {
     _value_type: PhantomData<V>,
```

Comment:
Consider adding a trait (or derive macro) for generating the imeplementations for binary operations. There's a lot of repeated boilerplate.

Reply:
Good idea. I've found a way to do it not only for binary operations